lib/opencensus/plug/trace.ex

```diff
@@ -112,13 +112,19 @@ defmodule Opencensus.Plug.Trace do
       :ocp.with_span_ctx(ctx)
     end
 
-    encoded =
-      :ocp.current_span_ctx()
-      |> :oc_span_ctx_header.encode()
-      |> :erlang.iolist_to_binary()
+    case :ocp.current_span_ctx() do
+      :undefined ->
```

Comment:
This should be impossible case.

Reply:
I agree it should be impossible, but my Phoenix project's tests fail when this plug is installed. If the pdict key used to store the context is not set, it returns `:undefined` and `:ocp_span_ctx_header.encode(:undefined)` raises a function-clause error.

Reply:
Note also the modified function is called at the top of the plug, before any call to set the child span.

Reply:
That is why this is bug. This part of the function should be moved to L85 as I mentioned earlier, which will end with span being present there.